### PR TITLE
KOGITO-2422 - Add support for Milestones in Data Index

### PR DESCRIPTION
**Pull request by cristianonicolai**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by nmirasch — APPROVED** [review]

File: data-index/data-index-storage/data-index-storage-api/src/main/java/org/kie/kogito/index/model/Milestone.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    private String id;
+    private String name;
+    private String status;
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
```
Comment (ruromero) [inline data-index/data-index-storage/data-index-storage-api/src/main/java/org/kie/kogito/index/model/Milestone.java:29]: can we make it fluent?

File: data-index/data-index-storage/data-index-storage-infinispan/src/test/java/org/kie/kogito/index/infinispan/protostream/MilestoneMarshallerTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        milestone.setStatus("status");
+
+        MessageMarshaller.ProtoStreamWriter writer = mock(MessageMarshaller.ProtoStreamWriter.class);
+
+        new MilestoneMarshaller(null).writeTo(writer, milestone);
+
+        InOrder inOrder = inOrder(writer);
+        inOrder.verify(writer).writeString("id", "id");
+        inOrder.verify(writer).writeString("name", "name");
+        inOrder.verify(writer).writeString("status", "status");
```
Comment (ruromero) [inline data-index/data-index-storage/data-index-storage-infinispan/src/test/java/org/kie/kogito/index/infinispan/protostream/MilestoneMarshallerTest.java]: Please use constants, will allow a better maintenance.

**Review by ruromero — CHANGES_REQUESTED** [review]
Just picky comments

**Comment by cristianonicolai** [comment]
@ruromero suggestions applied

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**